### PR TITLE
libcurl: Use absolute target names

### PR DESCRIPTION
**Pull request by franramirez688**
This PR updates the target names for CMakeDeps via set_property.
Conan version required: 1.43
Conan feature related: conan-io/conan#10099

### CODE REVIEW

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -536,9 +536,9 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "CURL")
-        self.cpp_info.set_property("cmake_target_name", "CURL")
+        self.cpp_info.set_property("cmake_target_name", "CURL::CURL")
```
Comment (SpaceIm) [inline recipes/libcurl/all/conanfile.py:539]:
```suggestion
        self.cpp_info.set_property("cmake_target_name", "CURL::libcurl")
```

File: recipes/libcurl/all/conanfile.py [hunk 2]
```diff
@@ -536,9 +536,9 @@ def package(self):
...
         self.cpp_info.set_property("pkg_config_name", "libcurl")
-        self.cpp_info.components["curl"].set_property("cmake_target_name", "libcurl")
+        self.cpp_info.components["curl"].set_property("cmake_target_name", "CURL::libcurl")
```
Comment (SpaceIm) [inline recipes/libcurl/all/conanfile.py:541]:
```suggestion
```

File: recipes/libcurl/all/conanfile.py [hunk 3]
```diff
@@ -536,9 +536,9 @@ def package(self):
...
         self.cpp_info.components["curl"].set_property("pkg_config_name", "libcurl")
```
Comment (SpaceIm) [inline recipes/libcurl/all/conanfile.py:542]: to remove in conan v2, side effect of `cmake_find_package_*` support

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/libcurl/all/conanfile.py (hunk 3 above)
Comment (franramirez688) [reply recipes/libcurl/all/conanfile.py]: Removed. The last changes for `pkg_config_name` define this property as an absolute name, so it'd be overwritten by the root one.

File: recipes/libcurl/all/conanfile.py [hunk 4]
```diff
@@ -536,10 +536,8 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "CURL")
-        self.cpp_info.set_property("cmake_target_name", "CURL")
+        self.cpp_info.set_property("cmake_target_name", "CURL::libcurl")
```
Comment (SpaceIm) [inline recipes/libcurl/all/conanfile.py:539]: `cmake_find_mode` should be set to `both` since there is a builtin FindCURL.cmake in CMake.

**Comment by SpaceIm** [comment]
@czoido How CMake variable names are prefixed in Find files generated by `CMakeDeps`? Here we want `CURL_FOUND`, `CURL_INCLUDE_DIRS`, `CURL_LIBRARIES` and `CURL_VERSION_STRING`: https://cmake.org/cmake/help/latest/module/FindCURL.html

**Comment by czoido** [comment]
> @czoido How CMake variable names are prefixed in Find files generated by `CMakeDeps`? Here we want `CURL_FOUND`, `CURL_INCLUDE_DIRS`, `CURL_LIBRARIES` and `CURL_VERSION_STRING`: https://cmake.org/cmake/help/latest/module/FindCURL.html

I have checked and CMakeDeps is not currently generating the correct variable names, it generates `libcurl_FOUND` and `libcurl_VERSION` so this is something that we must fix.
The `cmake_find_package` generator declared: `CURL_FOUND`, `CURL_VERSION`, `CURL_INCLUDE_DIRS` and `CURL_LIBRARIES`. I'll have a look at this.

**Comment by SpaceIm** [comment]
Here is a suggestion: https://github.com/conan-io/conan-center-index/pull/8425#issuecomment-993726762

**Review by danimtb — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**